### PR TITLE
Update adv_indexing decomposition for Llama 3B attention

### LINKED ISSUE #321: Update adv_indexing decomposition for Llama 3B attention

### PR DESCRIPTION
**Pull request by chandrasekaranpradeep**
Swap operands of embedding op in adv_index decompostion

Fixes #321

### CODE REVIEW

File: forge/test/mlir/test_ops.py [hunk 1]
```diff
@@ -558,3 +559,36 @@ def forward(self, x):
...
+        def __init__(self, name):
+            super().__init__(name)
+
+        def forward(self, input_tensor, indices):
+            return forge.op.AdvIndex("adv_index_op_1", input_tensor, indices)
+
+    model = ForgeAdvIndex("ForgeAdvIndex")
+
+    # Sample Inputs
+    pt_input_tensor = torch.rand((12, 100)).to(torch.float32)
```
Comment (nvukobratTT) [inline forge/test/mlir/test_ops.py:577]: Can we test out a bit more shape variants?

**Review by nvukobratTT — COMMENTED** [review]
As a note, let's not make it go to way to use ForgeModel. It'll be better to more rely on direct PyTorch and other frameworks. However, as this one is hard to map, it's okay to push it as is.

File: forge/test/mlir/test_ops.py (hunk 1 above)
Comment (chandrasekaranpradeep) [reply forge/test/mlir/test_ops.py:577]: Sure Nikola

**Review by nvukobratTT — APPROVED** [review]

**Review by dgolubovicTT — APPROVED** [review]

File: forge/forge/op/eval/forge/tm.py [hunk 2]
```diff
@@ -1167,7 +1167,7 @@ def decompose(type, attr, dc, inputs):
             # Consider the case adv_index(X,Y) where
             #    X: (A, B), Y: (1, C) or (C,) and A != 1
             if len(in0_shape) == 2:
-                result = dc.op("embedding", inputs,)
+                result = dc.op("embedding", (inputs[1], inputs[0]),)
```
[the remark this answers is not in the document]
Comment (chandrasekaranpradeep) [reply forge/forge/op/eval/forge/tm.py:1170]: Sure

[the remark this answers is not in the document]
Comment (dgolubovicTT) [reply forge/forge/op/eval/forge/tm.py:1173]: Awesome!

[the remark this answers is not in the document]
Comment (chandrasekaranpradeep) [reply forge/forge/op/eval/forge/tm.py:1173]: @dgolubovicTT  Added comment in the adv_index decompostion

**Pull request merged**